src/exec.c

```diff
@@ -185,6 +187,96 @@ struct hyper_event_ops err_ops = {
 	/* don't need write buff, the stderr data is one way */
 };
 
+int hyper_setup_exec_user(struct hyper_exec *exec)
+{
+	char *user = exec->user == NULL || strlen(exec->user) == 0 ? NULL : exec->user;
+	char *group = exec->group == NULL || strlen(exec->group) == 0 ? NULL : exec->group;
+
+	// check the config
+	if (!user) {
+		if (group || exec->nr_additional_groups > 0) {
+			fprintf(stderr, "group or additional groups can only be set when user is set\n");
```

Comment:
when any user(user, group, additional groups) config is set, the user field should be set.